### DISCUSSION TITLE
请问使用table异步请求分页时，第三方服务使用skip分页要怎么处理？

### DISCUSSION THREAD

**Discussion by iDeMonnnnnn** [layui/layui#2104] [问答] [answered]
### 议题条件

- [X] 我确认已查看官方使用文档：https://layui.dev ，但没有找到相关解决方案。
- [X] 我确认已在 [Issues](https://github.com/layui/layui/issues?q=is%3Aissue ) 中搜索过类似的问题，但没有找到相关解决方案。
- [x] 我已仔细阅读： 🍀[ Layui Issue 贡献指南](https://github.com/layui/layui/blob/main/CONTRIBUTING.md)

### 议题类型

功能请求

### 使用版本

2.9.14

### 问题描述

请问使用table异步请求分页时，第三方服务使用skip分页要怎么处理？

### 业务代码

```auto
url: "https://fitc7cds.lc-cn-n1-shared.com/1.1/classes/App",
        where: { skip: (curPage - 1) * pageCount },
```

### 浏览器

64

### 友好承诺

- [X] 我承诺将本着相互尊重、理解和友善的态度进行交流，共同维护 Layui 良好的社区氛围。

**Comment by sentsim** [comment, marked as answer]
借助 before 函数实现
参考：https://layui.dev/docs/2/table/#options.before

**Comment by iDeMonnnnnn** [comment]
var curPage = ??;
var pageCount = options.limit;
options.where.skip = (curPage - 1) * pageCount;

curPage为当前页码，请问要怎么拿到？

**Comment by iDeMonnnnnn** [comment]
收到别人的启发，直接读h5标签内容，解决了：
```
        before: function (options) {
            console.log(options); // 当前实例属性配置项
            var $ = layui.jquery;
            var curPage = $(".layui-laypage-skip").find("input").val(); //当前页码值
            //第一次没渲染完是undefined
            if (curPage == undefined) {
                curPage = 1;
            }
            console.log(curPage); // 当前实例属性配置项
            var pageCount = options.limit;
            options.where.skip = (curPage - 1) * pageCount; // 修改或额外追加 where 属性
        },
```

但是还是希望知道是否有一个属性 可以返回当前页码值？

**Reply by sentsim** [reply]
既然都打印了 `options`，为何不展开仔细看看它返回了些什么呢。

要好好加强下 Layui 的熟练度。

```js
before: function(options) {
  console.log(options.page.curr) // 当前页码值
},
```

**Comment by iDeMonnnnnn** [comment]
感谢，哈哈，我就是自己做一个工具才用开始写前端用Layui。
第一次before等于options.page.curr=undefined，我以为这里有问题。看来也是未渲染导致的。

![image](https://github.com/user-attachments/assets/89735175-e189-4524-be36-b7bacc0070fa)
